File: 28. Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.equals("")) return 0;
        int curr = 0;
        int n = needle.length();
        for(curr = 0; curr <= haystack.length() - n; curr++) {
            if (haystack.substring(curr, curr+n).equals(needle)) return curr;
        }
        return -1;
        
    }
}

File: amazon/python/28. Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if needle == "":
            return 0
        
        if len(haystack) < len(needle):
            return -1
        
        j = 0
        i = 0
        while i < len(haystack) - len(needle) + 1:
            if haystack[i] == needle[j]:
                cur = i
                while haystack[cur] == needle[j]:
                    j = j + 1
                    cur = cur + 1
                    if j == len(needle):
                        return cur - len(needle)
                j = 0
                i = i + 1
            else:
                i = i + 1
        return -1
    
    # Time: O(n) bad 0(N * M)
    # Space: O(1)
    
                
            
        
